SeQuant/core/eval_expr.cpp

```diff
@@ -91,7 +91,7 @@ Symmetry EvalExpr::infer_tensor_symmetry_prod(EvalExpr const& xpr1,
 
     if (ranges::distance(uniq_idxs) == tnsr1.const_braket().size()
                                      + tnsr2.const_braket().size()) {
-      return Symmetry::symm;
+      return Symmetry::antisymm;
```

Comment:
@bimalgaudel this seems arbitrary .... I looked at this whole function and it seems the logic is only equipped to handle the use cases that you have encountered so far?

Reply:
This function handles the cases of contraction between two tensors with any `sequant::Symmetry` combination. The outer product of the same tensor returns an antisymmetric tensor hence it appears arbitrary but it is not.

Reply:
@bimalgaudel can you explain this plz? Does not make sense to me.

Reply:
<img width="1279" alt="image" src="https://user-images.githubusercontent.com/44385776/162441242-6b81ef54-653f-400b-9689-a8741c639c0d.png">

This is a screenshot from OpMin paper by Pai-Wei Lai et al.

This function determines the symmetry of an intermediate when
1. bra indices (all of them) of tensor A are contracted with the ket indices (all of them) of tensor B.

    (symmetry of A, symmetry of B ) | symmetry of `contract(A,B)`
    --- | ---
    (Asymm, Asymm) | Asymm
    (Symm, Symm) | Symm
    (Symm, Asymm) or (Asymm, Symm)  | Symm (*this case appears to be missing from the code*)
    (*, *) | Nonsymm

2. an outer product of a tensor with itself, `outerproduct(A,A)`, results Asymm intermediate.